src/stacks/nestedBillingAlert.ts

```diff
@@ -0,0 +1,126 @@
+
+    const budgetParams: b.CfnBudgetProps = {
+      budget: {
+        budgetName: budgetName,
+        budgetType: 'COST',
+        budgetLimit: {
+          amount: alarmOpts.budgetLimit,
+          unit: 'USD',
+        },
+        timeUnit: 'MONTHLY',
+      },
+      notificationsWithSubscribers: [
+        {
+          notification: {
+            notificationType: 'ACTUAL',
+            comparisonOperator: 'GREATER_THAN',
+            threshold: 100,
+            thresholdType: 'PERCENTAGE',
+          },
+          subscribers: [
+            {
+              subscriptionType: 'SNS',
+              address: topicArn,
+            },
+          ],
+        },
+      ],
+    };
+
+    new b.CfnBudget(this, budgetName, budgetParams);
+  }
+
+  private createAnomalyDetectionAlarm(id: string, billingAlert: config.ConfigCustomBillingAlert, topic: string): void {
+    const topicArn = cdk.Token.asString(this.snsStack.getTopicArn(topic));
+    const conf = billingAlert.alarm?.[topic];
+
+    if (conf && conf.enabled !== false) {
+      const metricName = 'EstimatedCharges';
+      const anomalyDetectorName = `${id}-${topic}-anomaly-detector`;
+      const anomalyDetectorParams: cw.CfnAnomalyDetectorProps = {
+        metricName,
+        namespace: 'AWS/Billing',
+        stat: billingAlert.metric?.statistic || 'Maximum',
+        configuration: {
+          metricTimeZone: 'UTC',
+        },
+      };
+      new cw.CfnAnomalyDetector(this, anomalyDetectorName, anomalyDetectorParams);
+
+      const alarmName = `${anomalyDetectorName}-${anomalyDetectorParams.metricName}`;
+      const anomalyDetectorAlarmParams: cw.CfnAlarmProps = {
+        alarmName,
+        actionsEnabled: true,
+        ...conf,
+        alarmActions: [topicArn],
+        okActions: billingAlert.autoResolve ? [topicArn] : [],
+        thresholdMetricId: 'ad1',
+        treatMissingData: getTreatMissingData(conf?.treatMissingData),
+        comparisonOperator: getComparisonOperator(conf?.comparisonOperator),
+        metrics: [
+          {
+            expression: 'ANOMALY_DETECTION_BAND(m1, 1)',
+            id: 'ad1',
+          },
+          {
+            metricStat: {
+              metric: {
+                ...anomalyDetectorParams,
+                dimensions: [
+                  {
+                    name: 'Currency',
+                    value: 'USD',
+                  },
+                ],
+              },
+              stat: anomalyDetectorParams.stat,
+              period: getDuration(billingAlert.metric?.period).toSeconds(),
+            },
+            id: 'm1',
+          },
+        ],
+      };
+      new cw.CfnAlarm(this, `${anomalyDetectorName}-alarm`, anomalyDetectorAlarmParams);
+    }
+  }
+}
+
+export function createBillingAlertStack(stack: cdk.Stack, snsStack: NestedSNSStack): NestedBillingAlertStack {
+  return new NestedBillingAlertStack(stack, stack.stackName + '-billing-alert', snsStack);
```

Comment:
This function should check if there is any billing alerts to add. Otherwise it will error due to empty stack

Reply:
@nikolauska ,I've checked it without billing alerts config and it works fine. Could you specify the issue more?

Reply:
On the line 23 and 26 we are checking if the billing alerts are enabled. Now if I were to have this call in my monitoring cdk project, but disabled in the config it would create empty stack which would error out.

Check for example how lambda stack does this: https://github.com/nordcloud/mca-monitoring/blob/cdb6fc6616f51c27d15b4bc5eb81acd876fdc416/src/stacks/nestedLambda.ts#L56

Reply:
It seems CDK has fixed the issue with stack without any resources. Current version will just skip creating stack if there are no resources when earlier version would throw error. So this can be resolved